plugin-server/src/main/ingestion-queues/anonymous-event-buffer-consumer.ts

```diff
@@ -64,16 +64,25 @@ export const startAnonymousEventBufferConsumer = async ({
             const processEventAt = Number.parseInt(message.headers.processEventAt.toString())
             const now = Date.now()
             if (processEventAt > now) {
+                const eventId = message.headers.eventId.toString()
                 status.info('🔁', 'Delaying event processing', {
                     topic: batch.topic,
                     partition: batch.partition,
-                    eventId: message.headers.eventId.toString(),
+                    eventId: eventId,
                     delayMs: processEventAt - now,
                     processEventAt,
                     now,
                 })
                 const resume = pause()
```

Comment:
we're able to pause and resume things on a partition level, maybe we should do that?

Reply:
My thought  was that pause passed to eachBatch is a shortcut to `consumer.pause({topic, partition})` but I'll double check!

Reply:
Yeah I think this is correct according to https://kafka.js.org/docs/consuming#a-name-each-batch-a-eachbatch